src/schema/rules/tabular_data/nirs.yaml

```diff
@@ -0,0 +1,64 @@
+---
+nirsChannels:
+  selectors:
+    - datatype == "nirs"
+    - suffix == "channels"
+    - extension == ".tsv"
+  initial_columns:
+    - name__channels
+    - type__nirs_channels
+    - source__channels
+    - detector__channels
+    - wavelength_nominal
+    - units__nirs
+  columns:
+    name__channels: required
+    type__nirs_channels: required
+    source__channels: required
+    detector__channels: required
+    wavelength_nominal: required
+    units__nirs: required
+    sampling_frequency:
+      level: optional
+      level_addendum: required if `SamplingFrequency` is `n/a` in `_nirs.json`
+    orientation_component:
+      level: optional
+      level_addendum: required if `type` is `ACCEL`, `GYRO` or `MAGN`
+    wavelength_actual: optional
+    description: optional
+    wavelength_emission_actual: optional
+    short_channel: optional
+    status: optional
+    status_description: optional
+  additional_columns: allowed_if_defined
+
+nirsOptodes:
+  selectors:
+    - datatype == "nirs"
+    - suffix == "optodes"
+    - extension == ".tsv"
+  initial_columns:
+    - name__optodes
+    - type__optodes
+    - x__optodes
+    - y__optodes
+    - z__optodes
```

Comment:
Again, is this order meant to be enforced? And is `name` going to be unique?

Reply:
Yes, order should be enforced, I will add to spec text.
Name MUST be unique. I will add this to the spec text. Is there a way to make the validator enforce that name is unique?

Reply:
Yes, we'll implement it with #1304, so again, I don't think you need to do anything. I just wanted to make sure that my interpretation was correct.

Reply:
Thanks @effigies